src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/unit_tests/security/FakeSession.kt

```diff
@@ -0,0 +1,96 @@
+package org.vaccineimpact.orderlyweb.tests.unit_tests.security
+
+import java.util.*
+import javax.servlet.ServletContext
+import javax.servlet.http.HttpSession
+import javax.servlet.http.HttpSessionContext
+
+class FakeSession() : HttpSession
```

Comment:
It would be quite fiddly if not impossible to verify the correct behaviour with a mock as the logic reads from, writes to and updates entries from a list of attributes. Actually implementing a basic attributes store in this fake session object is a simple and accurate way of testing that after the method has run, the attributes are in the state you would expect.